pkg/asset/installconfig/azure/validation.go

```diff
@@ -92,6 +93,58 @@ func ValidateDiskEncryptionSet(client API, ic *types.InstallConfig) field.ErrorL
 	return allErrs
 }
 
+// ValidateSecurityProfileDiskEncryptionSet ensures the security profile disk encryption set exists and is valid.
+func ValidateSecurityProfileDiskEncryptionSet(client API, ic *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	invalidDiskEncryptionSetTypeMsg := fmt.Sprintf("the disk encryption set should be created with type %s", azenc.DiskEncryptionSetTypeConfidentialVMEncryptedWithCustomerKey)
+
+	if ic.Platform.Azure.DefaultMachinePlatform != nil &&
+		ic.Platform.Azure.DefaultMachinePlatform.OSDisk.SecurityProfile != nil &&
+		ic.Platform.Azure.DefaultMachinePlatform.OSDisk.SecurityProfile.DiskEncryptionSet != nil {
+		desFieldPath := field.NewPath("platform").Child("azure", "defaultMachinePlatform", "osDisk", "securityProfile", "diskEncryptionSet")
+		diskEncryptionSet := ic.Platform.Azure.DefaultMachinePlatform.OSDisk.SecurityProfile.DiskEncryptionSet
+		resp, err := client.GetDiskEncryptionSet(context.TODO(), diskEncryptionSet.SubscriptionID, diskEncryptionSet.ResourceGroup, diskEncryptionSet.Name)
```

Comment:
Can this block of code from L107 - L112 can be moved to a new method? It appears 3 times in this validation code.

Reply:
Valid point and it kind of smells to me too. But it's not exactly the same piece of code, it varies between the `default`, `control-plane` and `compute` definitions and I could not find a good way to refactor these 3 lines to a method. Any ideas that could help make this code more readable and sustainable?

Reply:
It appears that` desFieldPath` and `diskEncryptionSet` differ for `default`, `control-plane` and `compute`. Provide those as input parameters to a method that does the rest . You could move the definition of `invalidDiskEncryptionSetTypeMsg` into this method that returns error.
You could add the returned err from this method to allErrs using a single call before the return at L145.
`allErrs = append(allErrs, field.Invalid(desFieldPath, diskEncryptionSet, err.Error()))`

Reply:
Did you have something like [this](https://github.com/openshift/installer/pull/7469/files#diff-75606c90c997f4218689ce173a97316d61be39a8df4f5bebe08a4cd0e9f24973R96-R104) in mind?